app/src/test/kotlin/me/myshows/android/api/BaseMockWebServerTest.kt

```diff
@@ -0,0 +1,29 @@
+package me.myshows.android.api
+
+import okhttp3.mockwebserver.Dispatcher
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.After
+import org.junit.Before
+
+abstract class BaseMockWebServerTest {
+
+    protected val server: MockWebServer = MockWebServer().apply {
+        setDispatcher(object : Dispatcher() {
+            override fun dispatch(request: RecordedRequest): MockResponse = response(request)
+        })
+    }
+
+    @Before
+    fun setup() = server.start()
+
+    @After
+    fun stop() = server.shutdown()
+
+    protected fun resolve(path: String): String = server.url(path).toString()
+    protected fun notFound(): MockResponse = MockResponse().setResponseCode(404)
```

Comment:
Okhttp has no constants for HTTP response status? 
Just in case, we can use `HttpURLConnection.HTTP_NOT_FOUND`.